### ISSUE TITLE
Merging coverage files obtained from mapped paired-end and unmapped R1 and R2 reads 

### ISSUE THREAD

**Issue by shaghayeghsoudi** [FelixKrueger/Bismark#719]
Hi Felix,

For our FFPE samples I have called CpGs from both mapped paired end reads and unmapped reads (R1 and R2) to get the most out of our samples. Just to explain, we used Diagnose RRBS kit and did paired end sequencing on human tumour and normals.

I am merging coverage files for each specimen (Mapped paired end + unmapped R1 + unmapped R2) and I see there are some CpGs shared between the three groups. 

Here is an example 

```

> example1<-dd[dd$id=="1_567122",]
       V1     V2     V3         V4 V5 V6       id
8       1 567122 567122 100.000000  2  0 1_567122
109748  1 567122 567122   0.000000  0  5 1_567122
166047  1 567122 567122   4.166667  1 23 1_567122

> example2<-dd[dd$id=="1_567240",]
       V1     V2     V3 V4 V5 V6       id
14      1 567240 567240  0  0  4 1_567240
109753  1 567240 567240  0  0  1 1_567240

> example3<-dd[dd$id=="1_567206",]
       V1     V2     V3       V4 V5 V6       id
11      1 567206 567206 1.980198  2 99 1_567206
109751  1 567206 567206 0.000000  0  5 1_567206
166049  1 567206 567206 3.703704  1 26 1_567206
```

As you see for the same position there are variations. As I am merging these files for my downstream analysis I am wondering what would be the best way to deal with positions like these? Should I take the one came from mapped paired-end? I appreciate if you could provide some advice and thanks so much for developing awesome tools to process methylation data.

**Comment by FelixKrueger** [comment]
HI @shaghayeghsoudi 

This is tricky question, not sure there is one true answer to this. 

You could take take a pragmatic approach and simply merge all calls from all 3 coverage files (which would fit best with your approach of 'getting the most from your samples'). This comes with the potential drawback that you might end up with alignments from both R1 and R2 which overlap, and you would score these positions twice from single end alignments, but only once from paired-end alignments. In other words, you would up-weight calls from reads that don't align well in a paired-end setting which I don't think is desirable. On the other hand, the question is how often you encounter these situations (chances are that this is a negligible 'problem'). You assess this by trying to do a 'virtual Hi-C' approach of pairing up single-end alignments of R1 and R2, and handle the situation by excluding methylation calls from both single-end alignments if they fall on the same position (akin to what the overlap exclusion in paired-end mode does by default). Both of these approaches require quite a bit of custom work (as you really need to include the readIDs for this), and the question is: is it worth the effort?

Arguably, the simplest and most elegant option would be to only use the paired-end alignments, but that would call into question the rather painful approach of doing PE + 2 SE alignments in the first place. If you see that the vast majority of alignments and methylation calls comes from the PE setting, and only a tiny fraction is added by the SE bolt-on, I would probably do that as my preferred option. If you see a larger number of rescued reads coming from the SE approach, you are back to the 'how should I merge the methylation calls' from the first paragraph...

Does this help at all?

**Comment by shaghayeghsoudi** [comment]
Hi Felix, thanks for sharing your thoughts. Well I did something very simple :)
As nearly half of our samples are FFPEs, we did not get as much positions as Fresh frozen samples. So I had to do my best to get the most out of them.
If the position is shared between paired, R1 and R2, I simply took the one from paired-end alignment and if it is shared between R1 and R2 I took the one from R2 as it has better quality read. I am not sure if these approach makes sense at all!
